lib/core/src/Cardano/Wallet/DB/Sqlite.hs

```diff
@@ -569,6 +571,102 @@ migrateManually tr proxy defaultFieldValues =
+            TableMissing ->
+                traceWith tr $ MsgManualMigrationNotNeeded field
+            ColumnMissing -> do
+                -- there are no fee values, so we don't know anything about
+                -- the key deposit and must add a default value
+                addKeyDepositIfMissing
+
+                traceWith tr $ MsgManualMigrationNeeded field defaultValue
+                query <- Sqlite.prepare conn $ T.unwords
+                    [ "ALTER TABLE", tableName field
+                    , "ADD COLUMN", fieldName field
+                    , fieldType field, "NOT NULL"
+                    , "DEFAULT", defaultValue
+                    , ";"
+                    ]
+                _ <- Sqlite.step query
+                Sqlite.finalize query
+
+            -- adding key deposit in this case can be delayed, because we
+            -- can pull out the info from the fee values
+            ColumnPresent -> do
+                fee_policyInfo <- Sqlite.prepare conn $ T.unwords
+                    [ "SELECT"
+                    , fieldName field
+                    , "FROM"
+                    , tableName field
+                    , ";"
+                    ]
+                row <- Sqlite.step fee_policyInfo
+                    >> Sqlite.columns fee_policyInfo
+                Sqlite.finalize fee_policyInfo
+                case filter (/= PersistNull) row of
+                    [PersistText t] -> do
+                        traceWith tr $ MsgManualMigrationNeeded field t
+                        (a:b:rest) <- pure (T.splitOn " + " t)
+
+                        -- update fee policy
+                        let newVal = a <> " + " <> b
+                        query <- Sqlite.prepare conn $ T.unwords
+                            [ "UPDATE"
+                            , tableName field
+                            , "SET"
+                            , fieldName field
+                            , "= '" <> newVal <> "'"
+                            , ";"
+                            ]
+                        Sqlite.step query *> Sqlite.finalize query
+
+                        -- update stake key val
+                        case rest of
+                            [c] -> updateStakeKeyVal (DBField ProtocolParametersKeyDeposit) c
+                            _ -> pure ()
+
+                    [] -> addKeyDepositIfMissing
+                    xs -> fail ("Unexpected row result when querying fee value: " <> show xs)
+      where
+        -- | Adds a 'key_deposit column to the 'protocol_parameters'
+        -- table if it is missing.
+        --
+        addKeyDepositIfMissing :: IO ()
+        addKeyDepositIfMissing = do
+            addColumn_ conn True (DBField ProtocolParametersKeyDeposit) value
+          where
+            value = toText $ defaultKeyDeposit defaultFieldValues
+
+        updateStakeKeyVal :: DBField -> Text -> IO ()
+        updateStakeKeyVal stakeField c = do
```

Comment:
What about using a single function to cover all cases? Especially because they have quite different names although are acting on exactly the same table and column. `updateStakeKeyVal` should be enough in other cases too.

Reply:
I tried changing

```hs
    addColumn
        :: Sqlite.Connection
        -> Bool
        -> DBField
        -> Text
        -> Maybe (IO ()) -- ^ action to run when column is present
        -> IO SqlColumnStatus
    addColumn conn notNull field value updateCB = do
        isFieldPresent conn field >>= \st -> st <$ case st of
            TableMissing ->
                traceWith tr $ MsgManualMigrationNotNeeded field
            ColumnMissing -> do
                traceWith tr $ MsgManualMigrationNeeded field value
                query <- Sqlite.prepare conn $ T.unwords
                    [ "ALTER TABLE", tableName field
                    , "ADD COLUMN", fieldName field
                    , fieldType field, if notNull then "NOT NULL" else ""
                    , "DEFAULT", value
                    , ";"
                    ]
                _ <- Sqlite.step query
                Sqlite.finalize query
            ColumnPresent ->
                maybe (traceWith tr $ MsgManualMigrationNotNeeded field) id updateCB
```

but I think it makes it more convoluted. Although some code is shared here, it doesn't improve understanding imo.

Reply:
I improved the names a bit.